realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -451,28 +556,43 @@ protected final boolean checkAndReportInvalidState() {
             // TODO: Should we expose the underlying Realm errors directly? What else would make sense?
             boolean managementErrorHappened;
             boolean permissionErrorHappened;
+            boolean defaultPermissionErrorHappened;
             ObjectServerError managementError;
             ObjectServerError permissionError;
+            ObjectServerError defaultPermissionError;
             synchronized (permissionManager.errorLock) {
                 // Only hold lock while making a safe copy of current error state
                 managementErrorHappened = (permissionManager.managementRealmError != null);
                 permissionErrorHappened = (permissionManager.permissionRealmError != null);
+                defaultPermissionErrorHappened = (permissionManager.defaultPermissionsRealmError != null);
                 managementError = permissionManager.managementRealmError;
                 permissionError = permissionManager.permissionRealmError;
+                defaultPermissionError = permissionManager.defaultPermissionsRealmError;
             }
 
-            if (permissionErrorHappened && !managementErrorHappened) {
+            // Handle single errors
+            if (permissionErrorHappened && !managementErrorHappened && !defaultPermissionErrorHappened) {
                 notifyCallbackError(permissionError);
                 return true;
             }
 
-            if (managementErrorHappened && !permissionErrorHappened) {
+            if (managementErrorHappened && !permissionErrorHappened && !defaultPermissionErrorHappened) {
                 notifyCallbackError(managementError);
                 return true;
             }
 
-            if (permissionErrorHappened && managementErrorHappened) {
-                notifyCallbackError(combineRealmErrors(managementError, permissionError));
+            if (defaultPermissionErrorHappened && !permissionErrorHappened && !managementErrorHappened) {
+                notifyCallbackError(defaultPermissionError);
+            }
```

Comment:
lines 573 - 586 don't seems to be necessary. I think `combineRealmErrors()` could handle the single error case well.